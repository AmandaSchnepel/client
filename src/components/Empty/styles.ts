import styled from 'styled-components'
import { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
`

export const Image = styled.img`
  max-width: 35%;

  ${media.lessThan('small')`
    max-width: 80%;
`}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.black1};
    font-size: 2.4rem;
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: 1.4rem;
    font-weight: 400;
    margin-bottom: ${theme.spacings.medium};
  `}
`
