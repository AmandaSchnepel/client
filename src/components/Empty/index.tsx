import * as S from './styles'
import Button from 'components/Button'
import Link from 'next/link'

export type EmptyProps = {
  title: string
  description: string
  hasLink?: boolean
}

const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <S.Image src="/img/carrinhovazio.png" alt="Carrinho vazio" role="image" />

    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>

    {hasLink && (
      <Link href="/" passHref>
        <Button as="a">Voltar para a loja</Button>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
