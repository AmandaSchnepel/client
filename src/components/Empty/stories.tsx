import { Story, Meta } from '@storybook/react'
import Empty, { EmptyProps } from '.'

export default {
  title: 'Empty',
  component: Empty
} as Meta

export const Default: Story<EmptyProps> = (args) => <Empty {...args} />

Default.args = {
  title: 'Seu carrinho est√° vazio',
  description: 'Adicione produtos em seu carrinho',
  image: '/img/carrinhovazio.png'
}
