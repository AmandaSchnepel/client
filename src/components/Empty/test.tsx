import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

const props = {
  title: 'Seu carrinho está vazio',
  description: 'Uma simples descrição'
}

describe('<Empty />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('image', { name: /carrinho vazio/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Seu carrinho está vazio/i })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()

    expect(screen.getByText(/uma simples descrição/i)).toBeInTheDocument()
    ;('')
  })
})
