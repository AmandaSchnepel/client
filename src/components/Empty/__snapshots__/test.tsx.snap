// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 35%;
}

.c2 {
  color: #212529;
  font-size: 2.4rem;
}

.c3 {
  color: #31322D;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 3.2rem;
}

.c4 {
  background: #207824;
  color: #FFFFFF;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.55rem;
  height: 4rem;
  width: 28rem;
  font-size: 1.6rem;
}

@media (max-width:450px) {
  .c1 {
    max-width: 80%;
  }
}

<body>
  <div>
    <main
      class="c0"
    >
      <img
        alt="Carrinho vazio"
        class="c1"
        role="image"
        src="/img/carrinhovazio.png"
      />
      <h2
        class="c2"
      >
        Seu carrinho está vazio
      </h2>
      <p
        class="c3"
      >
        Uma simples descrição
      </p>
      <button
        class="c4"
      >
        <span>
          Voltar para a loja
        </span>
      </button>
    </main>
  </div>
</body>
`
